src/plugins/inspector/public/views/requests/components/details/req_code_viewer.tsx

```diff
@@ -39,19 +41,37 @@ const openInSearchProfilerLabel = i18n.translate('inspector.requests.openInSearc
 /**
  * @internal
  */
-export const RequestCodeViewer = ({ indexPattern, json }: RequestCodeViewerProps) => {
+export const RequestCodeViewer = ({
+  indexPattern,
+  requestParams,
+  json,
+}: RequestCodeViewerProps) => {
   const { services } = useKibana<InspectorPluginStartDeps>();
 
   const navigateToUrl = services.application?.navigateToUrl;
 
-  const devToolsDataUri = compressToEncodedURIComponent(`GET ${indexPattern}/_search\n${json}`);
+  function getValue() {
+    if (!requestParams) {
+      return json;
+    }
+
+    const fullPath = requestParams.querystring
+      ? `${requestParams.path}?${requestParams.querystring}`
+      : requestParams.path;
+
+    return `${requestParams.method} ${fullPath}\n${json}`;
+  }
+
+  const value = getValue();
+
+  const devToolsDataUri = compressToEncodedURIComponent(value);
   const consoleHref = services.share.url.locators
     .get('CONSOLE_APP_LOCATOR')
     ?.useUrl({ loadFrom: `data:text/plain,${devToolsDataUri}` });
   // Check if both the Dev Tools UI and the Console UI are enabled.
   const canShowDevTools =
     services.application?.capabilities?.dev_tools.show && consoleHref !== undefined;
-  const shouldShowDevToolsLink = !!(indexPattern && canShowDevTools);
+  const shouldShowDevToolsLink = !!(requestParams && canShowDevTools);
```

Comment:
Why have you removed the index pattern requirement from here? For ESQL?

Reply:
I removed indexPattern from shouldShowDevToolsLink because dev tools URL is no longer generated with indexPattern. 

The previous implementation was `const devToolsDataUri = compressToEncodedURIComponent(`GET ${indexPattern}/_search\n${json}`);` Now, the implementation no longer uses indexPattern and just uses requestParam.